example-app/cmd/app/main.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	_ "net/http/pprof" // nolint: gosec
+	"strconv"
+	"time"
+
+	zipkinExporter "contrib.go.opencensus.io/exporter/zipkin"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/lvl484/course-infra/example-app/api"
+	discovery "github.com/lvl484/course-infra/example-app/discovery/consul"
+	appMetrics "github.com/lvl484/course-infra/example-app/metrics"
+	openzipkin "github.com/openzipkin/zipkin-go"
+	zipkinHTTPReporter "github.com/openzipkin/zipkin-go/reporter/http"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+var (
+	version = "unknown"
+)
+
+func main() {
+	log.Print("start example.app")
+
+	appConfig := struct {
+		ServiceName   string `envconfig:"SERVICE_NAME"                   required:"true"`
+		BindIP        string `envconfig:"BIND_IP"                        required:"true"`
+		BindPort      int    `envconfig:"BIND_PORT"                      required:"true"`
+		BindDebugPort int    `envconfig:"BIND_DEBUG_PORT"`
+		Consul        struct {
+			Address string `envconfig:"CONSUL_ADDRESS"                   required:"true"`
+		}
+		Zipkin struct {
+			Address      string  `envconfig:"ZIPKIN_ADDRESS"             required:"true"`
+			ReportingAPI string  `envconfig:"ZIPKIN_REPORTING_API"       required:"true"`
+			Fraction     float64 `envconfig:"ZIPKIN_FRACTION"            required:"true"`
+		}
+	}{}
```

Comment:
Can take away to type
type AppConfig struct {......

Reply:
+1. But better make new file for this struct and add method to make one and return pointer